1-server-side/msal-webapi-ciam-sample/msal-web-api/src/main/java/com/microsoft/azure/msalapiciamsample/AADClaimsVerifier.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.azure.msalapiciamsample;
+
+import org.springframework.security.oauth2.common.exceptions.InvalidTokenException;
+import org.springframework.security.oauth2.provider.token.store.JwtClaimsSetVerifier;
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+
+public class AADClaimsVerifier implements JwtClaimsSetVerifier {
+    private static final String ISS_CLAIM = "iss";
+    private static final String AUD_CLAIM = "aud";
+    private static final String V2_ISSUER_FORMAT = "https://%s.ciamlogin.com/%s/v2.0";
+
+
+    private HashSet<String> acceptedIssuers = new HashSet<>();
+    private final String applicationId;
+
+    public AADClaimsVerifier(final String[] acceptedTenants, final String applicationId) {
+
+        Assert.notEmpty(acceptedTenants, "acceptedTenants cannot be empty");
+        for (final String tenant : acceptedTenants) {
+            Assert.notNull(tenant, "acceptedTenant value cannot be null");
+        }
+
+        Assert.notNull(applicationId, "applicationId (for audience claim) cannot be null");
+
+        this.applicationId = applicationId;
+
+        generateAcceptedIssuers(acceptedTenants);
+    }
+
+    private void generateAcceptedIssuers(final String[] acceptedTenants) {
+        for (String acceptedTenant : acceptedTenants) {
+            acceptedIssuers.add(String.format(V2_ISSUER_FORMAT, acceptedTenant, acceptedTenant));
+        }
+    }
+
+    /**
+     * This method is used by Spring to verify certain parts of the access token.
+     * <p>
+     * Currently, it ensures that the issue and audience exist, that the issuer matches the accepted issues defined
+     * in the security.oauth2.aad.aliases field of application.properites, and that the audience matches the API set up
+     * in Azure
+     */
+
+    public void verify(final Map<String, Object> claims) throws InvalidTokenException {
+        if (CollectionUtils.isEmpty(claims))
+            throw new InvalidTokenException("token must contain claims");
+        if (!claims.containsKey("iss"))
```

Comment:
nit: use the constant since you already defined one

Reply:
Fixed in latest commit.